File: src/Budget/Budget.Core/Models/Records/RecordModel.cs
using Budget.Core.Entities;
using Budget.Core.Models.Accounts;
using Budget.Core.Models.Categories;
using Budget.Core.Models.PaymentTypes;
using System;

namespace Budget.Core.Models.Records
{
    public class RecordModel
    {
        public int Id { get; set; }

        public string Note { get; set; }

        public AccountModel FromAccount { get; set; }

        public AccountModel Account { get; set; }

        public RecordType RecordType { get; set; }

        public PaymentTypeModel PaymentType { get; set; }

        public CategoryModel Category { get; set; }

        public DateTime DateCreated { get; set; }

        public DateTime RecordDate { get; set; }

        public decimal Amount { get; set; }

        public static RecordModel FromRecord(Record record)
        {
            return new RecordModel
            {
                Id = record.Id,
                Note = record.Note,
                Account = AccountModel.FromAccount(record.Account),
                FromAccount = record.FromAccountId.HasValue ? AccountModel.FromAccount(record.FromAccount) : null,
                RecordType = record.RecordType,
                PaymentType = PaymentTypeModel.FromPaymentType(record.PaymentType),
                Category = CategoryModel.FromCategory(record.Category),
                DateCreated = record.DateCreated,
                RecordDate = record.RecordDate,
                Amount = record.Amount,
            };
        }
    }
}

File: web/budget-web/src/app/shared/models/records/record.model.ts
import { AccountModel } from '../accounts/account.model';
import { CategoryModel } from '../categories/category.model';
import { PaymentTypeModel } from '../payment-types/payment-type.model';

export class RecordModel {
  public id: number;
  public note: string;
  public fromAccount: AccountModel;
  public account: AccountModel;
  public recordType: string;
  public paymentType: PaymentTypeModel;
  public category: CategoryModel;
  public dateAdded: Date;
  public recordDate: Date;
  public amount: number;
}
